src/gsy_e/models/strategy/forward/__init__.py

```diff
@@ -94,14 +119,122 @@ def event_offer_traded(self, *, market_id, trade):
+        """Apply the incoming live event to the strategy."""
+        self._live_event_handler.dispatch(event)
+
+
+class ForwardLiveEvents:
+    """Handle live events for the forward strategies."""
+
+    def __init__(self, strategy):
+        self._strategy = strategy
+
+    def dispatch(self, event: Dict):
+        """Apply a live event to the strategy."""
+        if not B2BLiveEvents.is_supported_event(event.get("type")):
+            self._strategy.log.error(
+                "Invalid event (%s) for area %s.", event, self._strategy.area.name)
+            return
+
+        if event.get("type") == B2BLiveEvents.ENABLE_TRADING_EVENT_NAME:
+            self._auto_trading_event(event.get("args"))
+        if event.get("type") == B2BLiveEvents.DISABLE_TRADING_EVENT_NAME:
+            self._stop_auto_trading_event(event.get("args"))
+        if event.get("type") == B2BLiveEvents.POST_ORDER_EVENT_NAME:
+            self._post_order_event(event.get("args"))
+        if event.get("type") == B2BLiveEvents.REMOVE_ORDER_EVENT_NAME:
+            self._remove_order_event(event.get("args"))
+
+    def _auto_trading_event(self, args):
+        """
+        Enable energy trading, and create order updater for all open markets between the start and
+        end time.
+        """
+        if not LiveEventArgsValidator(
+                self._strategy.log.error).validate_start_trading_event_args(args):
+            return
+
+        market_type = AvailableMarketTypes(args["market_type"])
+        start_time = str_to_pendulum_datetime(args["start_time"])
+        end_time = str_to_pendulum_datetime(args["end_time"])
+        capacity_percent = args["capacity_percent"]
+        energy_rate = args["energy_rate"]
+        market = self._strategy.area.forward_markets[market_type]
+
+        for slot in market.market_time_slots:
+            if not start_time <= slot <= end_time:
+                continue
+
+            if market not in self._strategy._order_updaters:
+                self._strategy._order_updaters[market] = {}
+            updater = self._strategy._order_updaters[market].get(slot)
+            if updater:
+                market.remove_open_orders(slot)
+
+            market_parameters = market.get_market_parameters_for_market_slot(slot)
+
+            order_updater_params = deepcopy(self._strategy._order_updater_params[market_type])
+            order_updater_params.capacity_percent = capacity_percent
+            order_updater_params.final_rate = energy_rate
+            order_updater_params.initial_rate = min(
+                order_updater_params.initial_rate, energy_rate)
+
+            self._strategy._order_updaters[market][slot] = OrderUpdater(
+                order_updater_params, market_parameters)
+            self._strategy.post_order(market, slot)
+
+    def _stop_auto_trading_event(self, event: Dict):
+        """Apply stop automatic trading event to the strategy."""
+        self._remove_order_event(event)
```

Comment:
This confused me at first. Do you agree on the following change?
```suggestion
    def _stop_auto_trading_event(self, args: Dict):
        """Apply stop automatic trading event to the strategy."""
        self._remove_order_event(args)
```

Reply:
Agreed, will rename this!